hugr-py/src/hugr/model/load.py

```diff
@@ -0,0 +1,1114 @@
+            node_id = self.add_node(
+                node, CFG(signature.input, signature.output), parent
+            )
+            self.import_cfg_region(body, signature, node_id)
+            return node_id
+
+        def import_conditional() -> Node:
+            match node.signature:
+                case model.Apply("core.fn", [inputs, outputs]):
+                    pass
+                case _:
+                    error = "Conditional node expects `(core.fn _ _)` signature."
+                    raise ModelImportError(error, node)
+
+            match list(_import_closed_list(inputs)):
+                case [model.Apply("core.adt", [variants]), *other_inputs]:
+                    sum_ty = Sum(
+                        [
+                            self.import_type_row(variant)
+                            for variant in _import_closed_list(variants)
+                        ]
+                    )
+                case _:
+                    error = (
+                        "Conditional node expects `(core.adt _)` as first input type."
+                    )
+                    raise ModelImportError(
+                        error,
+                        node,
+                    )
+
+            node_id = self.add_node(
+                node,
+                Conditional(
+                    sum_ty=sum_ty,
+                    other_inputs=[self.import_type(t) for t in other_inputs],
+                    _outputs=self.import_type_row(outputs),
+                ),
+                parent,
+            )
+
+            for case_body in node.regions:
+                case_signature = self.import_signature(case_body.signature)
+                case_id = self.hugr.add_node(
+                    Case(inputs=case_signature.input, _outputs=case_signature.output),
+                    node_id,
+                )
+                self.import_dfg_region(case_body, case_id)
+
+            return node_id
+
+        match node.operation:
+            case model.InvalidOp():
+                error = "Invalid operation can not be imported."
+                raise ModelImportError(error, node)
+            case model.Dfg():
+                return import_dfg_node()
+            case model.Cfg():
+                return import_cfg()
+            case model.Block():
+                error = "Unexpected basic block."
+                raise ModelImportError(error, node)
+            case model.CustomOp(op):
+                return import_custom_node(op)
+            case model.TailLoop():
+                return import_tail_loop()
+            case model.Conditional():
+                return import_conditional()
+            case _:
+                error = "Unexpected node in DFG region."
+                raise ModelImportError(error, node)
+
+    def import_node_in_module(self, node: model.Node, link_prefix: int) -> Node | None:
+        """Import a model Node at the Hugr Module level."""
+        self.link_prefix = link_prefix
```

Comment:
After all the hugrs are imported I think this should be unset. Could it be an explicit `enter`/`exit` paradigm like with `enter_symbol` instead of a field?